### ISSUE TITLE
Not all compatible files are showing up for wallpaper

### ISSUE THREAD

**Issue by RossComputerGuy** [pop-os/cosmic-settings#776]
Using commit 8a20cbc748c16eabf5bd3077c798e8cc3dd0975a, the screenshot shows the directory selected in the COSMIC Settings app and the same directory inside COSMIC Files but the settings app only shows 1 picture and not the 2nd picture.
![image](https://github.com/user-attachments/assets/99f46f14-1fb6-4f33-b6b1-668da8d3c6be)

**Comment by bhh32** [comment]
I'd like to take this if it's okay.

**Comment by bhh32** [comment]
@RossComputerGuy, can you upgrade? We're on commit 4e31002. If you upgrade, does it still do it? I can't seem to reproduce the issue.

**Comment by RossComputerGuy** [comment]
I have upgraded to 4e310024b7457921906df11bc042d40315f23de4 and it is still bugged.

**Comment by bhh32** [comment]
What are your system specs? I can't seem to recreate this on Pop!_OS 24.04, or Fedora 40.

**Comment by RossComputerGuy** [comment]
NixOS 24.11, Apple M1 Pro

**Comment by RossComputerGuy** [comment]
Yeah, I'm quite aware of that heh. I would try to reproduce this on my desktop but https://github.com/pop-os/cosmic-session/issues/92 prevents me from even getting me to log in.

**Comment by bhh32** [comment]
Isn’t the point of NixOS that you can rollback after something broke? Then
you can test with the latest version by updating your nixconfig?

**Comment by RossComputerGuy** [comment]
Yes but if I roll back then I don't have cosmic on that machine so I can't test.

**Comment by bhh32** [comment]
Yes, but you can update your config once you log in to have it again, just
with the newest version of everything. If you don’t want to do that, if you
get to the login screen, but not log in, you can press CTRL+ALT+ F2 and log
into a terminal session through the tty. Or, not sure if NixOS uses GRUB2,
but if it does you can press E on the GRUB screen and add a 3 at the end of
your kernel config to boot to a terminal session instead of a GUI session
and make your changes that way.
